packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -48,25 +48,73 @@ export function Comments( {
 	showCommentBoard,
 	setShowCommentBoard,
 } ) {
-	const { blockCommentId } = useSelect( ( select ) => {
-		const { getBlockAttributes, getSelectedBlockClientId } =
+	const { blockCommentId, blocks } = useSelect( ( select ) => {
+		const { getBlockAttributes, getSelectedBlockClientId, getBlocks } =
 			select( blockEditorStore );
 		const _clientId = getSelectedBlockClientId();
 
 		return {
 			blockCommentId: _clientId
 				? getBlockAttributes( _clientId )?.blockCommentId
 				: null,
+			blocks: getBlocks(),
 		};
 	}, [] );
 
+	const { selectBlock, clearSelectedBlock } = useDispatch( blockEditorStore );
+
 	const [ focusThread, setFocusThread ] = useState(
 		showCommentBoard && blockCommentId ? blockCommentId : null
 	);
 
-	const clearThreadFocus = () => {
-		setFocusThread( null );
-		setShowCommentBoard( false );
+	// Function to find and select blocks by comment ID
+	const selectBlocksByCommentId = ( commentId ) => {
+		if ( ! commentId || ! blocks ) return;
+
+		// Find blocks that have this comment ID
+		const relatedBlocks = [];
+		const findBlocks = ( blockList ) => {
+			blockList.forEach( ( block ) => {
+				if ( block.attributes?.blockCommentId === commentId ) {
+					relatedBlocks.push( block.clientId );
+				}
+				if ( block.innerBlocks ) {
+					findBlocks( block.innerBlocks );
+				}
+			} );
+		};
+
+		findBlocks( blocks );
+
+		// Select the first related block if found
+		if ( relatedBlocks.length > 0 ) {
+			selectBlock( relatedBlocks[ 0 ] );
+
+			const iframe = document.querySelector(
+				'iframe[name="editor-canvas"]'
```

Comment:
selecting and manipulating elements in the DOM this way feels fragile. Did you explore using a block attribute / datastore to tie the comment to the block/highlighting? 

Perhaps we could leverage the existing mechanism that ties between blocks and comments - clicking on a block highlights the related comment thread?